README
Date::Convert::French_Rev - Jean Forget <JFORGET@cpan.org>

Date conversion using the French Revolutionary calendar.

PRESENTATION

This   module  is   a  add-on   to  Mordechai   Abzug's  Date::Convert
module.  Since  his  module  is  pre-alpha,  mine  is,  too.   Another
consequence is that  this module allows you to convert  from or to any
calendar implemented in  Date::Convert.  These calendars are currently
Gregorian, Julian,  Hebrew and absolute  (which is Julian  Day Number,
with midnight-to-midnight days instead of noon-to-noon).

This module allows you to pretty-print French Revolutionary dates.

You can use this module even if you do not have Perl on your computer.
See the alternate install procedure below.

WHY A NEW VERSION AFTER TEN YEARS?

I released the supposedly last version  of this module (0.05) the same
day I  released DateTime::Calendar::FrenchRevolutionary  version 0.01.
It was intended to be  the final version of Date::Convert::French_Rev.
So why did I release version 0.06 ten years later?

Between  2003 and  2013, the  world has  moved on  and CPAN  has moved
on.  In  2003, UTF-8  was  a promising  technic,  but  not yet  widely
used. In  2013, UTF-8 *is*  widely used. In  2003, there was a  set of
CPAN  best  practices:  test   scripts  in  t/*.t  for  example.  Some
recommended practices appeared later: META.yml for example. And CPANTS
was  created  to  provide   a  static  code  analysis  checking  these
recommended practices. So  this new version is also  meant to catch up
with these  recommended practices. In  other words, I play  the CPANTS
game and I try to improve my rating by updating this module.

PREREQUISITES

Roman, by Alexandr Ciornii (CHORNY)
Date::Convert, by Mordechai Abzug (MORTY)

Optionally
Test::Exception, by Adrian Howard (ADIE)
Test::Warnings, by Karen Etheridge (ETHER)

INSTALL

First  intall  Roman.pm and  Date::Convert.pm,  and  then install  the
normal way:

perl Makefile.PL
make
make test
make install

Note: if "make  test" give you a "dubious"  result, one possibility is
that you  have forgotten to install  Roman.pm. I know,  it happened to
me!

Or use you favorite CPAN client: CPAN.pm, cpanp, cpanm, dzil, whatever.

ALTERNATE INSTALL

This module can be installed on a perl-less computer!

All you need, on this computer, is:
  - tar and gzip, or compatible software
  - a table-aware HTML browser (no Internet connection required if you
  downloaded the archive from a local network or a CD-ROM)
  - a printer.

The installation steps are:

- Gunzip and untar the archive.
- Go to the eg/result subdirectory.
- Open the various files ending in "en.html" (or, if you are a
French-speaker, ending in "fr.html") with the HTML browser.
- Print them.
That's all!


Copyright  (c) 2001-2003,  2013,  2015, 2020,  2021  Jean Forget.  All
rights reserved.  This program is  free software. You  can distribute,
modify, and otherwise mangle  Date::Convert::French_Rev under the same
terms as perl.


Changes
-*- encoding: utf-8; indent-tabs-mode: nil -*-
Revision history for Perl extension Date::Convert::French_Rev.

0.10  ????
        - new bugtracker: Github instead of rt.cpan.org

0.09  Primidi 1st Vendémiaire CCXXIX, jour du raisin, new year's day (Tuesday 22 September 2020)
        - fix a few typos in the feasts names (thank you Gérald)
        - add a few entries in the "SEE ALSO" chapter (thank you Gérald)
        - tidy up the documentation, for example replacing http:// by https:// whenever possible
        - start a deprecation process for the %L strftime specifier
        - check the scripts in the eg subdirectory

0.08  Tridi 23 Thermidor CCXXIII, jour de la lentille (Monday 2015-08-10,
      which, according to the Gregorian calendar, is the 223rd anniversary
      of the storming of the Tuileries palace)
        - fix prerequisite: Date::Convert 0.16 is fine, 0.14 is not.
          Actually, RT ticket 90470 was prematurely and erroneously declared
          as "fixed".
        - improve the documentation of method "change_to".
        - add more tests

0.07  Décadi 10 Thermidor CCXXIII, jour de l'arrosoir (Tuesday 2015-07-28, also
      the 221st anniversary of Robespierre's death)
        - fix prerequisite: Date::Convert 0.16 is fine, 0.14 is not (RT ticket 90470)
        - test file to check the parameter checks (mistakenly not included in 0.06)
        - a few documentation fixes
        - fix the mojibake bug (similar to RT ticket 100311 for DT::C::FR)
        - add an alternate API to convert from French Revolutionary to other calendar

0.06  Octidi 18 Brumaire CCXXII, jour de la dentelaire (Friday 2013-11-08, also
      the 214th anniversary of Bonaparte's coup agains the Directorate)
        - kwalitee issues: license, style, META.{yml,json}
        - use UTF-8 instead of ISO-8859-1, use spaces instead of tabs
        - more tests, more modern tests

0.05  Nonidi 9 Floréal CCXI, jour de la jacinthe (Monday 2003-04-28)
        - last version compatible with Date::Convert, next one will
        be in the DateTime::Calendar namespace, compatible with DateTime.pm
        - I had always believed that the additional days were not called
        Primidi, Duodi etc. But I have found a document where they are
        called Primidi etc. The bug in the module is fixed.

0.04  Tridi 13 Fructidor CCX, jour de l'épine-vinette (Friday 2002-08-30)
        - reorganise the directory structure along Matt Sergeant's tips
        - include the 4000-modulus rule for leap years (according to emacs-21.2)
        - reorganise the absolute → FR conversion
        - rephrase the README file
        - add the  prerequisite for Date::Convert  in Makefile.PL (the
        use statement and the @ISA array are not sufficient)
        - add a few examples in the eg/ subdirectory
        - fix a few bugs: $_ was squashed by date_string, the month value
        was not checked
        - compatibility with diagnostics-1.2

0.03  Décadi 20 Thermidor CCIX, jour de l'écluse (Tuesday 2001-08-07)
        - rewrite the Changes file, with proper dates
        - it seems that the 4th additional day should rather be "jour
        de l'opinion"
        - reformat the code

0.02  Nonidi 19 Prairial CCIX, jour du tilleul (Thursday 2001-06-07)
        - add the prerequisite for Roman.pm in Makefile.PL
        - extend the "SEE ALSO" section

0.01  Quartidi 4 Prairial CCIX, jour de l'angélique (Wednesday 2001-05-23)
        - original version; created by h2xs 1.19


Makefile.PL
#
#     Configuration script for Date::Convert::French_Rev
#     Copyright (c) 2001, 2002, 2003, 2013, 2015, 2020, 2021 Jean Forget
#
#     This program is distributed under the same terms as Perl 5.16.3:
#     GNU Public License version 1 or later and Perl Artistic License
#
#     You can find the text of the licenses in the F<LICENSE> file or at
#     L<https://dev.perl.org/licenses/artistic.html>
#     and L<https://www.gnu.org/licenses/gpl-1.0.html>.
#
#     Here is the summary of GPL:
#
#     This program is free software; you can redistribute it and/or modify
#     it under the terms of the GNU General Public License as published by
#     the Free Software Foundation; either version 1, or (at your option)
#     any later version.
#
#     This program is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#     GNU General Public License for more details.
#
#     You should have received a copy of the GNU General Public License
#     along with this program; if not, write to the Free Software Foundation,
#     Inc., <https://www.fsf.org/>.
#
use ExtUtils::MakeMaker;

WriteMakefile(
      'NAME'         => 'Date::Convert::French_Rev'
   ,  'VERSION_FROM' => 'lib/Date/Convert/French_Rev.pm'
   ,  'PREREQ_PM'    => {   Roman => 0
                          , Date::Convert   => '0.16'
                        }
   ,  'ABSTRACT'     => 'Date conversions using the French Revolutionary calendar'
   ,  'AUTHOR'       => 'Jean Forget <JFORGET@cpan.org>'
   ,  'LICENSE'      => 'perl'
   ,  'MIN_PERL_VERSION' => '5.8.8'
   ,  'BUILD_REQUIRES'   => { 'Test::More'         => '0',        # should be 'TEST_REQUIRES', except that ExtUtils::MakeMaker 6.5705 does not accept it
                        'ExtUtils::MakeMaker' => '6.57_02',
                           },
   , META_MERGE       => {
       'meta-spec' => { version => 2 },
       'prereqs' => {
          'build' => {
             'requires' => {
                'ExtUtils::MakeMaker' => '0',
             }
          },
          'configure' => {
             'requires' => {
                'ExtUtils::MakeMaker' => '0',
             }
          },
          'runtime' => {
             'requires' => {
                'Date::Convert'   => '0.16',
                'Roman'           => '0',
                'perl'            => '5.008008',
             }
          },
          'test' => {
             'recommends' => {
                  'Test::Exception' => '0'
                , 'Test::Warnings'  => '0'
             }
          },
       },
       provides       => {
           'Date::Convert::French_Rev' => {
                file => 'lib/Date/Convert/French_Rev.pm',
                version => '0.10',
           },
       },
       resources => {
           repository => {
               type => 'git',
               url  => 'https://github.com/jforget/Date-Convert-French_Rev.git',
               web  => 'https://github.com/jforget/Date-Convert-French_Rev',
           },
           bugtracker => {
               web  => 'https://github.com/jforget/Date-Convert-French_Rev/issues',
           },
       },
    },
);
